analysers/Analyser_Merge.py

```diff
@@ -409,14 +414,46 @@ def __init__(self, source, extractor = lambda json: json):
 
     def header(self):
         self.json = self.extractor(json.loads(self.source.open().read()))
-        return self.json[0].keys()
+        columns = map(lambda c: unaccent(c), self.json[0].keys())
+        return columns
```

Comment:
Why do you need to use unaccent ?

Reply:
Because one of the JSON dataset has a field named "Pôle", with the ô causing some encoding trouble.

Reply:
It should work. We already use lots raw column names.
Renaming the column is not in the spirit of the process. It's make the the code of the analyzer less understandable and more magic.

Reply:
Well, I am not a Python expert, maybe the encoding error thrown could be easily avoided. But using it as is wasn't working. And the input file seemed to be in UTF-8, which was making this error even more weird.

Reply:
Python with UTF-8 IS weird ! Ok I will look at that.

Reply:
Thank you :-)

Reply:
Which one contains "ô" ?

Reply:
Nantes file : http://data.nantes.fr/api/publication/24440040400129_NM_NM_00170/Toilettes_publiques_nm_STBL/content/?format=json

Reply:
```diff
diff --git a/analysers/Analyser_Merge.py b/analysers/Analyser_Merge.py
index eba6ce5..3564f0e 100644
--- a/analysers/Analyser_Merge.py
+++ b/analysers/Analyser_Merge.py
@@ -275,9 +275,6 @@ WHERE
     official.tags1 - osm_item.tags - 'source'::text != ''::hstore
 """
 
-def unaccent(s):
-   return ''.join(c for c in unicodedata.normalize('NFD', s) if unicodedata.category(c) != 'Mn')
-
 class Source:
     def __init__(self, attribution = None, millesime = None, url = None, name = None, encoding = "utf-8", file 
= None, fileUrl = None, fileUrlCache = 30, zip = None, filter = None):
         """
@@ -415,16 +412,15 @@ class JSON(Parser):
 
     def header(self):
         self.json = self.extractor(json.loads(self.source.open().read()))
-        columns = map(lambda c: unaccent(c), self.json[0].keys())
+        columns = self.json[0].keys()
         return columns
 
     def import_(self, table, srid, osmosis):
-        self.json = self.json or self.extractor(json.loads(self.source.open().read))
-        insert_statement = u"insert into %s (%%s) values %%s" % table
+        self.json = self.json or self.extractor(json.loads(self.source.open().read()))
         for row in self.json:
-            columns = map(lambda c: unaccent(c), row.keys())
-            values = map(lambda column: unicode(json.dumps(row[column])) if row[column] != None else None, row.keys())
-            osmosis.giscurs.execute(insert_statement, (psycopg2.extensions.AsIs(u",".join(map(lambda c: "\"%s\"" % c, columns))), tuple(values)))
+            osmosis.giscurs.execute(u"insert into \"%s\" (\"%s\") values (%s)" %
+                (table, u'", "'.join(row.keys()), (u'%s, ' * len(row.keys()))[:-2]),
+                map(json.dumps, row.values()))
 
 class GeoJSON(Parser):
     def __init__(self, source, extractor = lambda geojson: geojson):
```

Reply:
Thanks for the solution, I will commit this in the same time as flatten fix.